### ISSUE TITLE
When using the "Batch Count" buttons, all images are saved with the metadata for the first image

### ISSUE THREAD

**Issue by karl-von-knecht** [comfyanonymous/ComfyUI#1388]
To understand and reproduce: 
Let us look first at the case where it works correctly:
Use the attached (extremely basic) workflow and press the queue button five times.
You get 5 images.
Drag any of the 5 images back into ComfyUI and press queue.
Result: the image is reproduced, as expected.

Now the scenario where it fails:
Instead of pressing "queue" 5 times, check "Extra Options", set batch size to 5, then click "Queue" once.
Again, you get 5 images.
Reset batch size to 1.
Drag any of the 5 images back into ComfyUI and press queue.
Result: the first of the five images is reproduced, no matter which of the five you dragged in.
In other words: images 2,3,4 and 5 cannot be reproduced.
![ComfyUI_00001_](https://github.com/comfyanonymous/ComfyUI/assets/123109169/747cd2a9-29cf-4f80-ac32-6d23580a3429)
